### PR TITLE
Capture broadcasting deps in fusion

### PR DESCRIPTION
**Pull request by phofl**
@rjzamora I'd appreciate if you could take a look here, I am not too happy with he solution I found.

Background: Our fusion is not aggressive enough which results in a pegged scheduler for the tpch benchmarks on 1TB, this will get way worse if we re-run for 10TB. This PR achieves the same level of blockwise fusion that we have in dask/dask

### CODE REVIEW

**Comment by rjzamora** [comment]
Thanks! I'll definitely look through this today.

File: dask_expr/_expr.py [hunk 1]
```diff
@@ -2248,9 +2248,11 @@ def _fusion_pass(expr):
...
                 group.append(next)
                 for dep in dependencies[next]:
-                    if (dep.npartitions == root.npartitions) and not (
-                        dependents[dep] - set(stack) - set(group)
-                    ):
+                    if (
+                        dep.npartitions == root.npartitions
+                        or dep.npartitions == 1
+                        or root.npartitions == 1
+                    ) and not (dependents[dep] - set(stack) - set(group)):
```
Comment (rjzamora) [inline dask_expr/_expr.py:2255]:
```suggestion
                    if (
                        dep.npartitions == root.npartitions
                        or next._broadcast_dep(dep)
                    ) and not (dependents[dep] - set(stack) - set(group)):
```
Your logic makes sense, but I think we can simply check if `dep` is a "broadcastable" dependency of `next`.

**Review by rjzamora — COMMENTED** [review]
Your solution seems good to me @phofl - Just made some small suggestions that *may* make the code easier to follow.

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -2381,7 +2383,8 @@ def _task(self, index):
                 graph.update(subgraph)
                 graph[(name, index)] = name
             else:
-                graph[(_expr._name, index)] = _expr._task(index)
+                i = 0 if self._broadcast_dep(_expr) else index
+                graph[(_expr._name, i)] = _expr._task(i)
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_expr.py]: Sure

**Review by rjzamora — APPROVED** [review]

**Pull request merged**